### PR TITLE
docs: Bluetooth: Mesh: add Subnet Bridge doc

### PR DESCRIPTION
**Pull request by akredalen**
Added high-level documentation for Subnet Bridge,
connecting models together and explaining usage.

### CODE REVIEW

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+The Bridge Configuration Client can get the Subnet Bridge feature status of a node by sending a **Bridge Configuration Get** message to the Bridge Configuration Server model on the target node, using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_get` function.
+The Client can also request to get the list of subnets that are currently bridged by a node by sending a **Bridged Subnets Get** message by calling the :c:func:`bt_mesh_brg_cfg_cli_bridged_subnets_get` function.
+
+Bridging Table
+==============
+
+The Bridging Table contains the entries for the subnets that are brigded by the node, and is managed by the Bridge Configuration Server model.
+The Bridge Configuration Client can get the Bridging Table status by sending a **Bridging Table Get** message to the Server model on the target node, calling the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_get` function.
+
+The maximum number of subnets a node can bridge is defined by the :kconfig:option:`BT_MESH_BRG_TABLE_ITEMS_MAX` option, which defaults to the minimum value of 16, with a maximum possible size of 255.
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:48]:
```suggestion
The maximum number of subnets a node can bridge is defined by the :kconfig:option:`CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX` option, which defaults to the minimum value of 16, with a maximum possible size of 255.
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+Subnet Bridge
+=============
+
+A Subnet Bridge is a node in a Mesh network that connects multiple subnets, allowing communication between them by relaying messages across subnet groups.
+Only nodes with the Subnet Bridge functionality enabled can perform this role.
+The Subnet Bridge node has a primary subnet, based on the primary NetKey, which handles the IV Update procedure and propagates updates to other subnets.
+The secondary subnets, on which messages are relayed, are referred to as *bridged subnets*.
+The Bridge Configuration Client can get the Subnet Bridge feature status of a node by sending a **Bridge Configuration Get** message to the Bridge Configuration Server model on the target node, using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_get` function.
+The Client can also request to get the list of subnets that are currently bridged by a node by sending a **Bridged Subnets Get** message by calling the :c:func:`bt_mesh_brg_cfg_cli_bridged_subnets_get` function.
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:40]: You will probably want to add some breathing room as this is rendering as literally just one big paragraph (unless you insert blank lines), which is hard to visually "parse".
This comment applies to other blocks of text in your PR but this one seems to be the one where readability is impacted the most

Also, .rst files should wrap at 100 characters like all other files in Zephyr.

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+API reference
+*************
+
+This section contains types and defines common to the Bridge Configuration models.
+
+.. doxygengroup:: bt_mesh_brg
+   :project: Zephyr
+   :members:
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:76]:
```suggestion
.. doxygengroup:: bt_mesh_brg
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+Only nodes with the Subnet Bridge functionality enabled can perform this role.
+The Subnet Bridge node has a primary subnet, based on the primary NetKey, which handles the IV Update procedure and propagates updates to other subnets.
+The secondary subnets, on which messages are relayed, are referred to as *bridged subnets*.
+The Bridge Configuration Client can get the Subnet Bridge feature status of a node by sending a **Bridge Configuration Get** message to the Bridge Configuration Server model on the target node, using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_get` function.
+The Client can also request to get the list of subnets that are currently bridged by a node by sending a **Bridged Subnets Get** message by calling the :c:func:`bt_mesh_brg_cfg_cli_bridged_subnets_get` function.
+
+Bridging Table
+==============
+
+The Bridging Table contains the entries for the subnets that are brigded by the node, and is managed by the Bridge Configuration Server model.
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:45]:
```suggestion
The Bridging Table contains the entries for the subnets that are bridged by the node, and is managed by the Bridge Configuration Server model.
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+The Client can request the size of the Bridging Table from a Server by sending a **Bridging Table Size Get** message, using the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_size_get` function.
+
+Configuration and Management
+****************************
+
+The configuration and management of the Subnet Bridge feature are handled using the Bridge Configuration Server and Client models.
+A node implementing the Bridge Configuration Client model can act as a *Configuration Manager* for the Subnet Bridge feature.
+
+Enable or Disable the Subnet Bridge Feature
+===========================================
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:59]:
```suggestion
Enabling or Disabling the Subnet Bridge Feature
===============================================
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+The Bridge Configuration Client (or Configuration Manager) can enable or disable the Subnet Bridge feature on a node by sending a **Bridge Configuration Set** message to the Bridge Configuration Server model on the target node, using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_set` function.
+
+Add or Remove Subnets
+=====================
```
Comment (kartben) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:64]:
```suggestion
Adding or Removing Subnets
==========================
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by hermabe** [comment]
It would be good to get the Bluetooth MAINTAINERS.yml file changes into the mesh collab branch so that reviewers are assigned correctly.

**Review by PavelVPV — DISMISSED** [review]

**Review by PavelVPV — DISMISSED** [review]

**Comment by akredalen** [comment]
There is still some problem with the doc build. I will wait for @mia-ko to comment before attempting to resolve it.

**Review by kartben — COMMENTED** [review]
for the remaining warning, it might be best for you to just rebase on main, if possible (which you will have to do anyway when it will be time to merge the collab branch), to make it go away

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+The Bridge Configuration Client model is optional and allows nodes to configure the Subnet Bridge on
+other nodes.
+These models define the necessary states, messages, and behaviors for configuring and managing the
+Subnet Bridge feature.
+
+Concepts
+========
+
+The Subnet Bridge feature introduces several concepts to enable message relaying between different
+subnets in a Mesh network.
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:27]: To get a better understanding of the Subnet Bridge feature and its capabilities, an overview of a few concepts is needed.

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+Adding or Removing Subnets
+--------------------------
+
+The Bridge Configuration Client can add or remove subnets from a node's Bridging Table by sending a
+**Bridge Configuration Add** or **Bridge Configuration Remove** message to the Bridge Configuration
+Server model on the target node, calling the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_add` or
+:c:func:`bt_mesh_brg_cfg_cli_bridging_table_remove` functions.
+
+Relay and Replay Protection
+===========================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:100]:
```suggestion
Subnet Bridging and Replay Protection
=====================================
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+The Bluetooth Mesh Subnet Bridge feature allows a node to relay messages between different subnets
+in a Mesh network.
+
+The Subnet Bridge feature includes two models:
+
+.. toctree::
+   :maxdepth: 1
+
+   brg_cfg_srv
+   brg_cfg_cli
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:15]:
```suggestion
* :ref:`bluetooth_mesh_models_brg_cfg_srv`
* :ref:`bluetooth_mesh_models_brg_cfg_cli`
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
I was wondering, should we have a small section on subnet bridging and directed forwarding?

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 10]
```diff
@@ -0,0 +1,144 @@
+.. _bluetooth_mesh_brg_cfg:
+
+Subnet Bridge
+#############
+
+The Bluetooth Mesh Subnet Bridge feature allows a node to relay messages between different subnets
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:6]: I do not see that this is added to other files, so I will leave it as is for now.

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+The Bridge Configuration Client can get the Subnet Bridge feature status of a node by sending a
+**Bridge Configuration Get** message to the Bridge Configuration Server model on the target node,
+using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_get` function.
+
+The Client can also request to get the list of subnets that are currently bridged by a node by
+sending a **Bridged Subnets Get** message by calling the
+:c:func:`bt_mesh_brg_cfg_cli_bridged_subnets_get` function.
+
+Bridging Table
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:56]: Consider adding a States section covering all states:
* Subnet Bridge state
* Bridging Table state
* Bridging Table Size state

Comment (akredalen) [reply doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:57]: I made a new section. Please have a look.

**Review by PavelVPV — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+**Bridging Table Add** or **Bridging Table Remove** message to the Bridge Configuration
+Server model on the target node, calling the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_add` or
+:c:func:`bt_mesh_brg_cfg_cli_bridging_table_remove` functions.
+
+Subnet Bridge States
+====================
+
+The Subnet Bridge has the following states:
+
+- *Subnet Bridge Status*: Indicates whether the Subnet Bridge feature is enabled or disabled on
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:87]:
```suggestion
- *Subnet Bridge*: Indicates whether the Subnet Bridge feature is enabled or disabled on
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+  the node.
+  The Bridge Configuration Client can retrieve this by sending a **Subnet Bridge Get** message to
+  the Bridge Configuration Server using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_get`
+  function.
+
+- *Bridging Table Status*: The Client can request a list of entries from a Bridging Table by
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:93]:
```suggestion
- *Bridging Table*: The Client can request a list of entries from a Bridging Table by
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+
+- *Bridging Table Size*: Specifies the maximum number of entries the Bridging Table can store.
+  The Client can retrieve this information by sending a **Bridging Table Size Get** message using
+  the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_size_get` function.
+
+Subnet Bridging and Replay Protection
+=====================================
+
+The Subnet Bridge feature enables message relaying between subnets and requires effective replay
+protection to ensure network security. Key considerations include:
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:109]:
```suggestion
protection to ensure network security. Key considerations to take into account are described below.
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+This feature allows mesh networks to use subnets for area isolation, but to also allow communication
+between specific devices in different, adjacent subnets without compromising security.
+
+The Bluetooth Mesh Subnet Bridge feature makes it possible for selected nodes in the network to act
+as Subnet Bridges, allowing controlled communication by relaying messages between nodes in adjacent
+subnets.
+
+The Subnet Bridge feature includes two models:
+
+* :ref:`bluetooth_mesh_models_brg_cfg_srv`
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:17]: I see we now have both * and - for bullet point in the same file.
I would consider aligning on one of them, not that it has any effect for the rendering (it should still be rendered the same) :)

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by PavelVPV — DISMISSED** [review]

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+These models define the necessary states, messages, and behaviors for configuring and managing the
+Subnet Bridge feature.
+
+The configuration and management of the Subnet Bridge feature are handled using the Bridge
+Configuration Server and Client models.
+A node implementing the Bridge Configuration Client model can act as a *Configuration Manager* for
+the Subnet Bridge feature.
+
+Concepts
+========
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:32]:
```suggestion
Concepts
********
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+
+To get a better understanding of the Subnet Bridge feature and its capabilities, an overview of a
+few concepts is needed.
+
+Subnet
+------
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:38]:
```suggestion
Subnet
======
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+
+A subnet is a group of nodes within a mesh network that share a common network key, enabling them to
+communicate securely at the network layer.
+Each subnet operates independently, with nodes exchanging messages exclusively within that group.
+A node may belong to multiple subnets at the same time.
+
+Subnet Bridge node
+------------------
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:46]:
```suggestion
Subnet Bridge node
==================
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+a Subnet Bridge functionality enabled. Subnet bridging can be performed only by such a node. The
+Subnet Bridge node connects the subnets, and allows communication between them by relaying
+messages across the subnet groups.
+
+The Subnet Bridge node has a primary subnet, based on the primary NetKey, which handles the
+IV Update procedure and propagates updates to other subnets.
+The secondary subnets, on which messages are relayed, are referred to as *bridged subnets*.
+
+Bridging Table
+--------------
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:58]:
```suggestion
Bridging Table
==============
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+
+The Bridging Table contains the entries for the subnets that are bridged by the node, and is managed
+by the Bridge Configuration Server model.
+
+The maximum number of entries in the bridging table is defined by the
+:kconfig:option:`CONFIG_BT_MESH_BRG_TABLE_ITEMS_MAX` option, which defaults to the minimum value of
+16, with a maximum possible size of 255.
+
+Enabling or disabling the Subnet Bridge feature
+===============================================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:68]:
```suggestion
Enabling or disabling the Subnet Bridge feature
***********************************************
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+
+The Bridge Configuration Client (or Configuration Manager) can enable or disable the Subnet Bridge
+feature on a node by sending a **Subnet Bridge Set** message to the Bridge Configuration
+Server model on the target node, using the :c:func:`bt_mesh_brg_cfg_cli_subnet_bridge_set` function.
+
+Adding or removing subnets
+==========================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:75]:
```suggestion
Adding or removing subnets
**************************
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+
+The Bridge Configuration Client can add or remove an entry from the Bridging Table by sending a
+**Bridging Table Add** or **Bridging Table Remove** message to the Bridge Configuration
+Server model on the target node, calling the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_add` or
+:c:func:`bt_mesh_brg_cfg_cli_bridging_table_remove` functions.
+
+.. _bluetooth_mesh_brg_cfg_states:
+
+Subnet Bridge states
+====================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:85]:
```suggestion
Subnet Bridge states
********************
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 23]
```diff
@@ -0,0 +1,160 @@
...
+  **Bridged Subnets Get** message to the target Server by calling the
+  :c:func:`bt_mesh_brg_cfg_cli_bridged_subnets_get` function.
+
+- *Bridging Table Size*: This state reports the maximum number of entries the Bridging Table can
+  store. The Client can retrieve this information by sending a **Bridging Table Size Get** message
+  using the :c:func:`bt_mesh_brg_cfg_cli_bridging_table_size_get` function.
+  This is a read-only state.
+
+Subnet bridging and replay protection
+=====================================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:109]:
```suggestion
Subnet bridging and replay protection
*************************************
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+
+The Subnet Bridge feature enables message relaying between subnets and requires effective replay
+protection to ensure network security. Key considerations to take into account are described below.
+
+Relay buffer considerations
+---------------------------
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:115]:
```suggestion
Relay buffer considerations
===========================
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 25]
```diff
@@ -0,0 +1,160 @@
...
+When a message is relayed between subnets by a Subnet Bridge, it is allocated from the relay buffer.
+To ensure that messages can be retransmitted to all subnetworks,
+the :kconfig:option:`CONFIG_BT_MESH_RELAY_BUF_COUNT` option should be increased accordingly.
+
+However, if the :kconfig:option:`CONFIG_BT_MESH_RELAY` feature is disabled, the messages will be
+allocated from the advertising buffer instead. In this case, increase the
+:kconfig:option:`CONFIG_BT_MESH_ADV_BUF_COUNT` option to allow for sufficient buffer space.
+
+Replay protection and Bridging Table
+------------------------------------
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:126]:
```suggestion
Replay protection and Bridging Table
====================================
```

File: doc/connectivity/bluetooth/api/mesh/brg_cfg.rst [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+as every bridged message must pass through the replay protection mechanism before being relayed.
+
+.. note::
+
+   The RPL size should scale with the Bridging Table. As the number of bridged subnets grows,
+   more source addresses and IVISeq values must be tracked, requiring a larger RPL to maintain
+   effective replay protection.
+
+Subnet Bridge and Directed Forwarding
+=====================================
```
Comment (mia-ko) [inline doc/connectivity/bluetooth/api/mesh/brg_cfg.rst:149]:
```suggestion
Subnet Bridge and Directed Forwarding
*************************************
```

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Review by Balaklaka — APPROVED** [review]

**Comment by Balaklaka** [comment]
@kartben Do you have any more comments to this PR, or can you approve/remove your change request?

**Review by kartben — APPROVED** [review]

**Comment by PavelVPV** [comment]
@carlescufi , this PR can be merged now.

**Pull request merged**